### ISSUE TITLE
VoxelGridPhantom.build(): torch UserWarning to use sourceTensor.clone().detach()

### ISSUE THREAD

**Issue by o-dietrich** [MRsources/MRzero-Core#48]
In voxel_grid_phantom.py:146 (build()), pytorch (version 2.2.1+cu121) suggests to use clone():

>     /.../lib/python3.11/site-packages/MRzeroCore/phantom/voxel_grid_phantom.py:146: 
>     UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or 
>     sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
>     torch.tensor(shape, device=self.PD.device) / 2 / self.size,
 
so, maybe, 
>       shape.clone().detach() / 2 / self.size,
instead of
>       torch.tensor(shape, device=self.PD.device) / 2 / self.size,
 would work without warning? (Didn't check the device and if a copy is really necessary, here.)

**Comment by J-Endres** [comment]
Yes, this should work - I think there are also a few other deprecation and / or copy warnings. None of these are an error or at places critical for performance but definitely shold be fixed.
Leaving this open for now to collect the other places where this occurs.

**Comment by o-dietrich** [comment]
Agreed - another one might be in lines 108..118 of voxel_grid_phantom.py:

>        pos_x, pos_y, pos_z = torch.meshgrid(
>            self.size[0] *
>            torch.fft.fftshift(torch.fft.fftfreq(
>                int(shape[0]), device=self.PD.device)),
>            self.size[1] *
>            torch.fft.fftshift(torch.fft.fftfreq(
>                int(shape[1]), device=self.PD.device)),
>            self.size[2] *
>            torch.fft.fftshift(torch.fft.fftfreq(
>                int(shape[2]), device=self.PD.device)),
>        )

which should now have an additional parameter `indexing='ij'`.

**Comment by J-Endres** [comment]
A similar UserWarning is found in voxel_grid_phantom - line 146:
Tensor copying with `new = torch.tensor(old)` is discouraged, use `new = old.clone().detach()`
